### PR TITLE
Reduce SD Logging soak test duration & fix problem with multiple nodes

### PR DESCRIPTION
**Pull request by crassirostris**
Fixes https://github.com/kubernetes/kubernetes/issues/49012

/cc @piosz

### CODE REVIEW

File: test/e2e/instrumentation/logging/sd_soak.go [hunk 1]
```diff
@@ -53,9 +52,9 @@ var _ = instrumentation.SIGDescribe("Cluster level logging implemented by Stackd
 
 		nodes := framework.GetReadySchedulableNodesOrDie(f.ClientSet).Items
 		maxPodCount := 10
-		jobDuration := 1 * time.Hour
+		jobDuration := 30 * time.Minute
 		linesPerPodPerSecond := 100
-		testDuration := 21 * time.Hour
+		testDuration := 3 * time.Hour
```
Comment (piosz) [inline test/e2e/instrumentation/logging/sd_soak.go]: Could you please add a todo to revert o 21 hours?

File: test/e2e/instrumentation/logging/sd_soak.go [hunk 2]
```diff
@@ -65,19 +64,25 @@ var _ = instrumentation.SIGDescribe("Cluster level logging implemented by Stackd
 		linesPerPod := linesPerPodPerSecond * int(jobDuration.Seconds())
 
 		pods := []*loggingPod{}
+		podsByRun := [][]*loggingPod{}
 		for runIdx := 0; runIdx < podRunCount; runIdx++ {
+			podsInRun := []*loggingPod{}
```
Comment (piosz) [inline test/e2e/instrumentation/logging/sd_soak.go]: Could you please add a comment to explain the logic here?

**Comment by crassirostris** [comment]
@piosz Done, PTAL

**Pull request merged**